apps/frontend/src/components/ComponentsList/ComponentsListGroups.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react';
+import { pascalToSpaced } from '@react-site-editor/functions';
+import type { ComponentInfos } from '@react-site-editor/types';
+import Icon from '@components/Decorators/Icon';
+import Draggable from '@components/Decorators/Draggable';
+import ComponentsListItem from '@components/ComponentsList/ComponentsListItem';
+
+interface ComponentsListGroupsProps {
+    elements: ComponentInfos[];
+    filter: string;
+}
+
+const ComponentsListGroups: React.FunctionComponent<ComponentsListGroupsProps> = (props) => {
+    const filteredComponents = props.elements.filter((component) => {
+        return (
+            component.name.toLowerCase().includes(props.filter.toLowerCase()) ||
+            component.group.toLowerCase().includes(props.filter.toLowerCase())
+        );
+    });
+
+    const groups = [...new Set(filteredComponents.map((element) => element.group))];
+
+    const elementsByGroup: Record<string, ComponentInfos[]> = {};
+
+    filteredComponents.forEach((element) => {
+        if (!elementsByGroup[element.group]) {
+            elementsByGroup[element.group] = [];
+        }
+
+        elementsByGroup[element.group].push(element);
+    });
+
+    const Group: React.FunctionComponent<{ group: string }> = (groupProps) => {
+        const [visibility, setVisibility] = useState(false);
```

Comment:
Use this patterns for the boolean isVisible or isIconVisible

Reply:
Oh yes, my bad.

(old code, btw)